python/ml4ir/base/pipeline.py

```diff
@@ -188,48 +175,6 @@ def set_seeds(self, reset_graph=True):
         tf.random.set_seed(self.args.random_state)
         random.seed(self.args.random_state)
 
-    def validate_args(self):
-        """
-        Validate the arguments to be used with RelevancePipeline
-        """
-        unset_arguments = {key: value for (key, value) in vars(self.args).items() if value is None}
```

Comment:
we should be able to have None in arguments. This blocks this. Now every check is enforced in the argparser. Another benefit is we get more detailed help messages that are dynamic (we add something to the metrics, the metrics will directly appear).